### PR TITLE
[Refactor]Migrate all scheduler DAO methods to ScheduleDao

### PR DESCRIPTION
**Pull request by DarkAssassinator**
## Purpose of the pull request

+ Clean code
+ This is a part of https://github.com/apache/dolphinscheduler/issues/12403

## Brief change log

Migrate all schedule dao methods to new SchrduleDao

## Verify this pull request

This pull request is already covered by existing tests

### CODE REVIEW

**Review by EricGao888 — DISMISSED** [review]

**Comment by DarkAssassinator** [comment]
hi @EricGao888 could please help to merge?

**Comment by EricGao888** [comment]
> hi @EricGao888 could please help to merge?

CI blocked in doc check step. Could u plz rebase on the latest dev branch?

**Comment by EricGao888** [comment]
@DarkAssassinator 
image

**Comment by DarkAssassinator** [comment]
> @DarkAssassinator image

hi @EricGao888 , may i ask what this mean?

**Comment by DarkAssassinator** [comment]
hi @EricGao888, PTAL again, thx

File: dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/repository/ScheduleDao.java [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ * Schedule DAO
+ */
+public interface ScheduleDao {
+
+    /**
+     * Query schedule {@link Schedule} by id
+     * @param id schedule id
+     * @return schedule
+     */
+    Schedule querySchedule(int id);
```
Comment (ruanwenjun) [inline dolphinscheduler-dao/src/main/java/org/apache/dolphinscheduler/dao/repository/ScheduleDao.java:34]:
```suggestion
    Schedule queryScheduleById(int id);
```

File: dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java [hunk 2]
```diff
@@ -265,6 +269,7 @@ public WorkflowExecuteRunnable(
         this.taskInstanceDao = taskInstanceDao;
         this.taskDefinitionLogDao = taskDefinitionLogDao;
         this.masterAddress = NetUtils.getAddr(masterConfig.getListenPort());
+        this.scheduleDao = SpringApplicationContext.getBean(ScheduleDao.class);
```
Comment (ruanwenjun) [inline dolphinscheduler-master/src/main/java/org/apache/dolphinscheduler/server/master/runner/WorkflowExecuteRunnable.java:272]: Please don't use `SpringApplicationContext`, this class might be removed.

**Review by ruanwenjun — COMMENTED** [review]
Basically LGTM

**Pull request closed without merging**